x-pack/solutions/search/plugins/search_playground/public/providers/form_provider.tsx

```diff
@@ -75,13 +85,19 @@ export const FormProvider: React.FC<React.PropsWithChildren<FormProviderProps>>
     setValue: form.setValue,
     getValues: form.getValues,
   });
+  useUserQueryValidations({
+    watch: form.watch,
+    setValue: form.setValue,
+    getValues: form.getValues,
+  });
 
+  const setLocalSessionDebounce = useDebounceFn(setLocalSession, LOCAL_STORAGE_DEBOUNCE_OPTIONS);
   useEffect(() => {
     const subscription = form.watch((values) =>
-      setLocalSession(values as Partial<ChatForm>, storage)
+      setLocalSessionDebounce.run(values as PartialChatForm, storage)
```

Comment:
Even before adding query editing this was spamming local storage, adding debounce slows that down considerably.